misc/scripts/deploy-edgeX-snaps.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+timestamp() {
+  local ts=""
+  ts=$(date +"%Y-%m-%dT%T.%3N") # current time
+  echo "$ts -"
+}
+
+channel_type() {
+  local CHANNEL_VAL=""
+
+  if [[ -z "$1" ]]; then
+    echo "Value not set"
+  else
+    case $1 in
+      stable)
+        CHANNEL_VAL="--stable"
+        ;;
+
+      edge)
+        CHANNEL_VAL="--edge"
+        ;;
+
+      beta)
+        CHANNEL_VAL="--beta"
+      ;;
+
+      candidate)
+        CHANNEL_VAL="--candidate"
+      ;;
+    esac
+  fi
+
+  echo "$CHANNEL_VAL"
+}
+
+EF_CHANNEL_VAL=""
+EF_CHANNEL_VAL_SET=false
+if  [[ -n "$EF_CHANNEL" ]]; then
+  EF_CHANNEL_VAL="$(channel_type "${EF_CHANNEL}")"
+
+  if [[ -z "${EF_CHANNEL_VAL}" ]]; then
+    EF_CHANNEL_VAL_SET=false
+  else
+    EF_CHANNEL_VAL_SET=true
+  fi
+fi
+
+ASC_CHANNEL_VAL=""
+ASC_CHANNEL_VAL_SET=false
+if [[ -n "$ASC_CHANNEL" ]]; then
+  ASC_CHANNEL_VAL="$(channel_type "${ASC_CHANNEL}")"
+
+  if [[ -z "${ASC_CHANNEL_VAL}" ]]; then
+    ASC_CHANNEL_VAL_SET=false
+  else
+    ASC_CHANNEL_VAL_SET=true
+  fi
+fi
+
+EF_SNAP_VAL=""
+EF_SNAP_VAL_SET=false
+EF_SNAP_VAL_VALID=false
+
+if [[ -n "$EF_SNAP" ]]; then
+    EF_SNAP_VAL_SET=true
+    EF_SNAP_VAL=${EF_SNAP}
+
+    # Check if the edgexfoundry snap exists on filesystem
+    if test -f "$EF_SNAP_VAL"; then
+        EF_SNAP_VAL_VALID=true
+    fi
+fi
+
+ASC_SNAP_VAL=""
+ASC_SNAP_VAL_SET=false
+ASC_SNAP_VAL_VALID=false
+
+if [[ -n "$ASC_SNAP" ]]; then
+    ASC_SNAP_VAL_SET=true
+    ASC_SNAP_VAL=${ASC_SNAP}
+
+    # Check if the edgexfoundry snap exists on filesystem
+    if test -f "$ASC_SNAP_VAL"; then
+        ASC_SNAP_VAL_VALID=true
+    fi
+fi
+
+echo "============================================================================================"
+echo "                      Deploying edgex snap test service for Blackbox"
+echo "============================================================================================"
+
+echo "$(timestamp) Removing edgexfoundry snap (if already installed)..."
+sudo snap remove edgexfoundry
```

Comment:
I would also use `--purge` so as to not save a snapshot here.

Reply:
Added flag.